### Importing the necessary libraries

In [1]:
import pandas as pd
import numpy as np
import plotly.graph_objects as go

### Importing relevant datasets

In [2]:
df_sleep = pd.read_csv('Data/sleepDay_merged.csv')
df_activity = pd.read_csv('Data/dailyActivity_merged.csv')
df_weight = pd.read_csv('Data/weightLogInfo_merged.csv')

### Data Cleaning

#### Sleep

In [3]:
print(df_sleep.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 413 entries, 0 to 412
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   Id                  413 non-null    int64 
 1   SleepDay            413 non-null    object
 2   TotalSleepRecords   413 non-null    int64 
 3   TotalMinutesAsleep  413 non-null    int64 
 4   TotalTimeInBed      413 non-null    int64 
dtypes: int64(4), object(1)
memory usage: 16.3+ KB
None


In [4]:
#Convert SleepDay to datetime
df_sleep['SleepDay'] = pd.to_datetime(df_sleep['SleepDay'])

In [5]:
df_sleep.duplicated().sum()

3

In [6]:
#Drop duplicates
df_sleep.drop_duplicates(inplace=True)

In [7]:
df_sleep.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 410 entries, 0 to 412
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype         
---  ------              --------------  -----         
 0   Id                  410 non-null    int64         
 1   SleepDay            410 non-null    datetime64[ns]
 2   TotalSleepRecords   410 non-null    int64         
 3   TotalMinutesAsleep  410 non-null    int64         
 4   TotalTimeInBed      410 non-null    int64         
dtypes: datetime64[ns](1), int64(4)
memory usage: 19.2 KB


#### Activity

In [8]:
df_activity.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 940 entries, 0 to 939
Data columns (total 15 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Id                        940 non-null    int64  
 1   ActivityDate              940 non-null    object 
 2   TotalSteps                940 non-null    int64  
 3   TotalDistance             940 non-null    float64
 4   TrackerDistance           940 non-null    float64
 5   LoggedActivitiesDistance  940 non-null    float64
 6   VeryActiveDistance        940 non-null    float64
 7   ModeratelyActiveDistance  940 non-null    float64
 8   LightActiveDistance       940 non-null    float64
 9   SedentaryActiveDistance   940 non-null    float64
 10  VeryActiveMinutes         940 non-null    int64  
 11  FairlyActiveMinutes       940 non-null    int64  
 12  LightlyActiveMinutes      940 non-null    int64  
 13  SedentaryMinutes          940 non-null    int64  
 14  Calories  

In [9]:
#convert activity date to datetime
df_activity['ActivityDate'] = pd.to_datetime(df_activity['ActivityDate'])

In [10]:
#Create new column TotalActivityMins being the sum of VeryActiveMinutes, FairlyActiveMinutes, LightlyActiveMinutes 
df_activity['TotalActivityMins'] = df_activity['VeryActiveMinutes'] + df_activity['FairlyActiveMinutes'] + df_activity['LightlyActiveMinutes']

In [11]:
df_activity.describe()

Id    TotalSteps  TotalDistance  TrackerDistance  \
count  9.400000e+02    940.000000     940.000000       940.000000   
mean   4.855407e+09   7637.910638       5.489702         5.475351   
std    2.424805e+09   5087.150742       3.924606         3.907276   
min    1.503960e+09      0.000000       0.000000         0.000000   
25%    2.320127e+09   3789.750000       2.620000         2.620000   
50%    4.445115e+09   7405.500000       5.245000         5.245000   
75%    6.962181e+09  10727.000000       7.712500         7.710000   
max    8.877689e+09  36019.000000      28.030001        28.030001   

       LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
count                940.000000          940.000000                940.000000   
mean                   0.108171            1.502681                  0.567543   
std                    0.619897            2.658941                  0.883580   
min                    0.000000            0.000000                  0.000000   
25%                    0.000000            0.000000                  0.000000   
50%                    0.000000            0.210000                  0.240000   
75%                    0.000000            2.052500                  0.800000   
max                    4.942142           21.920000                  6.480000   

       LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
count           940.000000               940.000000         940.000000   
mean              3.340819                 0.001606          21.164894   
std               2.040655                 0.007346          32.844803   
min               0.000000                 0.000000           0.000000   
25%               1.945000                 0.000000           0.000000   
50%               3.365000                 0.000000           4.000000   
75%               4.782500                 0.000000          32.000000   
max              10.710000                 0.110000         210.000000   

       FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  \
count           940.000000            940.000000        940.000000   
mean             13.564894            192.812766        991.210638   
std              19.987404            109.174700        301.267437   
min               0.000000              0.000000          0.000000   
25%               0.000000            127.000000        729.750000   
50%               6.000000            199.000000       1057.500000   
75%              19.000000            264.000000       1229.500000   
max             143.000000            518.000000       1440.000000   

          Calories  TotalActivityMins  
count   940.000000         940.000000  
mean   2303.609574         227.542553  
std     718.166862         121.776307  
min       0.000000           0.000000  
25%    1828.500000         146.750000  
50%    2134.000000         247.000000  
75%    2793.250000         317.250000  
max    4900.000000         552.000000

In [12]:
df_activity.duplicated().sum()

0

#### Weight

In [13]:

df_weight.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 67 entries, 0 to 66
Data columns (total 8 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Id              67 non-null     int64  
 1   Date            67 non-null     object 
 2   WeightKg        67 non-null     float64
 3   WeightPounds    67 non-null     float64
 4   Fat             2 non-null      float64
 5   BMI             67 non-null     float64
 6   IsManualReport  67 non-null     bool   
 7   LogId           67 non-null     int64  
dtypes: bool(1), float64(4), int64(2), object(1)
memory usage: 3.9+ KB


Fat percentage is recorded for only 2 users. So dropping that column from our EDA. 

In [14]:
#Drop Fat column from df_weight
df_weight.drop('Fat', axis=1, inplace=True)

In [15]:
#Convert Date to datetime
df_weight['Date'] = pd.to_datetime(df_weight['Date'])

In [16]:
#check for duplicates in df_weight
df_weight.duplicated().sum()

0

In [17]:
df_weight.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 67 entries, 0 to 66
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   Id              67 non-null     int64         
 1   Date            67 non-null     datetime64[ns]
 2   WeightKg        67 non-null     float64       
 3   WeightPounds    67 non-null     float64       
 4   BMI             67 non-null     float64       
 5   IsManualReport  67 non-null     bool          
 6   LogId           67 non-null     int64         
dtypes: bool(1), datetime64[ns](1), float64(3), int64(2)
memory usage: 3.3 KB


In [18]:
#Split Date in df_weight into year, month, and day and time
df_weight['time'] = df_weight['Date'].dt.time
df_weight['date'] = df_weight['Date'].dt.date


In [19]:
#Add a column describing weight lost by each user
df_weight.sort_values(by=['Id', 'Date'], inplace=True)
df_weight['WeightLost'] = df_weight.groupby('Id')['WeightPounds'].transform('first') - df_weight.groupby('Id')['WeightPounds'].transform('last')
df_weight[['Id','WeightLost']].groupby('Id').first()

WeightLost
Id                    
1503960366    0.000000
1927972279    0.000000
2873212765   -1.322770
4319703577    0.220459
4558609924    1.322770
5577150313    0.000000
6962181067    1.322770
8877689391    3.968327

In [20]:
df_weight.columns

Index(['Id', 'Date', 'WeightKg', 'WeightPounds', 'BMI', 'IsManualReport',
       'LogId', 'time', 'date', 'WeightLost'],
      dtype='object')

### Identifying Trends in the dataset. 

#### Sleep


In [21]:
#Number of unique users in sleep data
print(f"Number of Unique Users: {df_sleep['Id'].nunique()}")
#Number of sleep records per user
print(f"Sleep records per user:\n{df_sleep['Id'].value_counts()}")


Number of Unique Users: 24
Sleep records per user:
8378563200    31
6962181067    31
5553957443    31
2026352035    28
3977333714    28
4445114986    28
4702921684    27
5577150313    26
4319703577    26
1503960366    25
7086361926    24
4388161847    23
6117666160    18
8792009665    15
2347167796    15
4020332650     8
1927972279     5
4558609924     5
1644430081     4
6775888955     3
8053475328     3
1844505072     3
7007744171     2
2320127002     1
Name: Id, dtype: int64


1) We see that some users have more than one sleep record per day. Not sure why this is the case, but let's ignore that anamoly and consider them to have entered data for 30 days. 
2) Only 24 out of 33 users have been tracking their sleep
3) Let's see how many users have tracked it consistently

In [22]:
#Number of users who have sleep records for atleast 15 days
print(f"Number of users who have sleep records for atleast 15 days: {df_sleep['Id'].value_counts()[df_sleep['Id'].value_counts() >= 15].count()}")

Number of users who have sleep records for atleast 15 days: 15


In [23]:
#Percentage of users tracking sleep for atleast 15 days out of 33 users
tracking_sleep_ok = df_sleep['Id'].value_counts()[df_sleep['Id'].value_counts() >= 15].count()/df_activity['Id'].nunique()*100

#Percentage of users tracking sleep for less than 15 days out of 33 users
tracking_sleep_inconsistent = df_sleep['Id'].value_counts()[df_sleep['Id'].value_counts() < 15].count()/df_activity['Id'].nunique()*100

#Percentage of users tracking not tracking their sleep at all
not_tracking = (df_activity['Id'].nunique() - df_sleep['Id'].nunique())/df_activity['Id'].nunique()*100

In [24]:
#Visualize the above
labels = ['Tracked Sleep for atleast 15 days', 'Tracked Sleep for less than 15 days', 'Did not track sleep at all']
values = [tracking_sleep_ok, tracking_sleep_inconsistent, not_tracking]
fig = go.Figure(data=[go.Pie(labels=labels, values=values)])
fig.update_layout(title_text='Sleep Tracking')
fig.show()


Majority of users ( 54.6 %) are either not tracking their sleep at all or are tracking inconsistently.
1. Is it a bother to sleep with it or is the sleep hours not tracked properly by the product?
2. Or are they charging the watch at night?
Not enough data to come to a conclusion.


In [25]:
#correlation between activity and sleep
df_sleep_activity = pd.merge(df_sleep, df_activity, left_on=['Id', 'SleepDay'], right_on=['Id', 'ActivityDate'], how='inner')
df_sleep_activity[['TotalSteps', 'TotalDistance','Calories', 'SedentaryMinutes','TotalActivityMins','TotalMinutesAsleep', 'TotalTimeInBed']].corr()
col = ['TotalSteps', 'TotalDistance','Calories', 'SedentaryMinutes', 'TotalActivityMins', 'TotalMinutesAsleep', 'TotalTimeInBed']

In [26]:
#heatmap of correlation between activity and sleep
fig = go.Figure(data=go.Heatmap(x = col, y = col, z=df_sleep_activity[col].corr().values, colorscale= 'Viridis',text=df_sleep_activity[col].corr().values, hoverinfo='text'))
fig.update_layout(title='Correlation between activity and sleep')
fig.show()

1. The more time a user spends in active mode, the more steps and distance a user covers.
2. The more TotalSteps the user takes, the more TotalDistance they cover. And the more distance the user covers, the more calories they burn. 
3. The more the sedantary minutes, the less time they spend in bed and sleeping. 
4. But that doesn't mean the moe active people are spending a healthy amount of time asleep or not. There's not much of a correlation between TotalActivityMins and TotalMinutesAsleep.

In [27]:
#Relationship between TotalActivityMins and sleep time
fig = go.Figure()
fig.add_trace(go.Scatter(x=df_sleep_activity['TotalActivityMins'], y=df_sleep_activity['TotalMinutesAsleep'], mode='markers', name='TotalMinutesAsleep'))
fig.add_trace(go.Scatter(x=df_sleep_activity['TotalActivityMins'], y=df_sleep_activity['TotalTimeInBed'], mode='markers', name='TotalTimeInBed'))
fig.update_layout(title='Relationship between TotalActivityMins and sleep time', xaxis_title='TotalActivityMins', yaxis_title='Sleep Time')
fig.show()

In [28]:
#Relationship between sedantary minutes and sleep time
fig = go.Figure()
fig.add_trace(go.Scatter(x=df_sleep_activity['SedentaryMinutes'], y=df_sleep_activity['TotalMinutesAsleep'], mode='markers', name='TotalMinutesAsleep'))
fig.add_trace(go.Scatter(x=df_sleep_activity['SedentaryMinutes'], y=df_sleep_activity['TotalTimeInBed'], mode='markers', name='TotalTimeInBed'))
fig.update_layout(title='Relationship between sedantary minutes and sleep time', xaxis_title='SedantaryMinutes', yaxis_title='SleepTime')
fig.show()


We can see the negative relationship between Sedantary Minutes and Sleep Time. The correlation between TotalMinutesAsleep and Sedantary minutes is -0.65.
Could this mean that Users with more sedantary minutes spend less time asleep ( hence poor sleep quality)?


In [29]:
df_weight.shape

(67, 10)

In [30]:
df_activity.shape

(940, 16)

In [31]:
#Merge df_activity and df_weight grouped on Id mean of TotalSteps, TotalDistance, Calories, SedentaryMinutes, TotalActivityMins, and mean of WeightPounds, WeightLossPounds

df_activity_weight = pd.merge(df_activity.groupby('Id').mean(), df_weight.groupby('Id').mean(), left_on='Id', right_on='Id', how='inner')
df_activity_weight.head()


TotalSteps  TotalDistance  TrackerDistance  \
Id                                                         
1503960366  12116.741935       7.809677         7.809677   
1927972279    916.129032       0.634516         0.634516   
2873212765   7555.774194       5.101613         5.101613   
4319703577   7268.838710       4.892258         4.892258   
4558609924   7685.129032       5.080645         5.080645   

            LoggedActivitiesDistance  VeryActiveDistance  \
Id                                                         
1503960366                       0.0            2.858387   
1927972279                       0.0            0.095806   
2873212765                       0.0            0.676129   
4319703577                       0.0            0.278065   
4558609924                       0.0            0.549355   

            ModeratelyActiveDistance  LightActiveDistance  \
Id                                                          
1503960366                  0.794194             4.152903   
1927972279                  0.031290             0.507097   
2873212765                  0.276129             4.143548   
4319703577                  0.502258             3.768710   
4558609924                  0.682258             3.847742   

            SedentaryActiveDistance  VeryActiveMinutes  FairlyActiveMinutes  \
Id                                                                            
1503960366                 0.000000          38.709677            19.161290   
1927972279                 0.000000           1.322581             0.774194   
2873212765                 0.005161          14.096774             6.129032   
4319703577                 0.000000           3.580645            12.322581   
4558609924                 0.000000          10.387097            13.709677   

            LightlyActiveMinutes  SedentaryMinutes     Calories  \
Id                                                                
1503960366            219.935484        848.161290  1816.419355   
1927972279             38.580645       1317.419355  2172.806452   
2873212765            308.000000       1097.193548  1916.967742   
4319703577            228.774194        735.806452  2037.677419   
4558609924            284.967742       1093.612903  2033.258065   

            TotalActivityMins    WeightKg  WeightPounds        BMI  \
Id                                                                   
1503960366         277.806452   52.599998    115.963147  22.650000   
1927972279          40.677419  133.500000    294.317120  47.540001   
2873212765         328.225806   57.000000    125.663489  21.570001   
4319703577         244.677419   72.350002    159.504452  27.415000   
4558609924         309.064516   69.639999    153.529918  27.214000   

            IsManualReport         LogId  WeightLost  
Id                                                    
1503960366             1.0  1.462277e+12    0.000000  
1927972279             0.0  1.460510e+12    0.000000  
2873212765             1.0  1.462190e+12   -1.322770  
4319703577             1.0  1.461672e+12    0.220459  
4558609924             1.0  1.461974e+12    1.322770

In [32]:
#Relationship between sedantary minutes and weight in pounds
fig = go.Figure()
fig.add_trace(go.Scatter(x=df_activity_weight['SedentaryMinutes'], y=df_activity_weight['WeightPounds'], mode='markers', name='WeightPounds'))
fig.update_layout(title='Relationship between sedantary minutes and weight', xaxis_title='SedantaryMinutes', yaxis_title='WeightPounds')
fig.show()


In [33]:
#Number of days users sleeping more than 6 hours
df_sleep['Number of days with qualty sleep'] = df_sleep[df_sleep['TotalMinutesAsleep'] >= 360].value_counts('Id')

#### Weight

In [34]:
#Number of unique IDs in df_weight
df_weight['Id'].nunique()

8

In [35]:
#Visualize weight lost by each user
fig = go.Figure()
fig.add_trace(go.Bar(x=df_weight['Id'].unique().astype(str), y=df_weight[['Id','WeightLost']].groupby('Id').first()['WeightLost']))
fig.update_layout(title='Weight lost by each user', xaxis_title='User ID', yaxis_title='WeightLost')
fig.show()


In [36]:
df_weight.sample(10)

Id                Date   WeightKg  WeightPounds        BMI  \
28  6962181067 2016-04-28 23:59:59  61.200001    134.922906  23.889999   
26  6962181067 2016-04-25 23:59:59  61.700001    136.025217  24.100000   
5   4319703577 2016-04-17 23:59:59  72.400002    159.614681  27.450001   
56  8877689391 2016-04-28 06:50:03  85.099998    187.613382  25.490000   
44  8877689391 2016-04-13 06:55:00  84.900002    187.172464  25.410000   
8   4558609924 2016-04-25 23:59:59  70.300003    154.984977  27.459999   
53  8877689391 2016-04-25 06:40:16  85.400002    188.274775  25.559999   
36  6962181067 2016-05-06 23:59:59  61.500000    135.584291  24.000000   
40  6962181067 2016-05-10 23:59:59  62.099998    136.907061  24.240000   
19  6962181067 2016-04-18 23:59:59  61.200001    134.922906  23.889999   

    IsManualReport          LogId      time        date  WeightLost  
28            True  1461887999000  23:59:59  2016-04-28    1.322770  
26            True  1461628799000  23:59:59  2016-04-25    1.322770  
5             True  1460937599000  23:59:59  2016-04-17    0.220459  
56           False  1461826203000  06:50:03  2016-04-28    3.968327  
44           False  1460530500000  06:55:00  2016-04-13    3.968327  
8             True  1461628799000  23:59:59  2016-04-25    1.322770  
53           False  1461566416000  06:40:16  2016-04-25    3.968327  
36            True  1462579199000  23:59:59  2016-05-06    1.322770  
40            True  1462924799000  23:59:59  2016-05-10    1.322770  
19            True  1461023999000  23:59:59  2016-04-18    1.322770

In [37]:
df_activity.sample(10)

Id ActivityDate  TotalSteps  TotalDistance  TrackerDistance  \
398  4319703577   2016-04-29        7990           5.36             5.36   
111  1844505072   2016-05-01        2573           1.70             1.70   
806  8253242879   2016-04-19        4212           2.78             2.78   
144  1927972279   2016-05-03        1326           0.92             0.92   
582  5577150313   2016-04-27        6805           5.14             5.14   
608  6117666160   2016-04-23       11495           8.68             8.68   
821  8378563200   2016-04-15       14461          11.47            11.47   
715  7007744171   2016-04-16        4631           3.10             3.10   
399  4319703577   2016-04-30        8221           5.52             5.52   
791  8053475328   2016-05-05       14990          11.51            11.51   

     LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
398                       0.0                0.45                      0.79   
111                       0.0                0.00                      0.26   
806                       0.0                0.00                      0.00   
144                       0.0                0.73                      0.00   
582                       0.0                1.81                      0.40   
608                       0.0                0.00                      0.00   
821                       0.0                4.91                      1.15   
715                       0.0                0.00                      0.00   
399                       0.0                0.40                      1.61   
791                       0.0                8.85                      0.45   

     LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
398                 4.12                      0.0                  6   
111                 1.45                      0.0                  0   
806                 2.78                      0.0                  0   
144                 0.18                      0.0                 10   
582                 2.93                      0.0                 63   
608                 8.68                      0.0                  0   
821                 5.41                      0.0                 60   
715                 3.10                      0.0                  0   
399                 3.51                      0.0                  6   
791                 2.21                      0.0                 93   

     FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  Calories  \
398                   18                   289               624      2175   
111                    7                    75               585      1541   
806                    0                   164              1276      1763   
144                    0                    17              1413      2195   
582                   16                   190               773      3294   
608                    0                   512               468      2651   
821                   23                   190               729      3666   
715                    0                   203              1155      2076   
399                   38                   196               695      2092   
791                    9                   130              1208      2950   

     TotalActivityMins  
398                313  
111                 82  
806                164  
144                 27  
582                269  
608                512  
821                273  
715                203  
399                240  
791                232

In [38]:
df_sleep.sample(10)

Id   SleepDay  TotalSleepRecords  TotalMinutesAsleep  \
60   2026352035 2016-05-08                  1                 541   
87   3977333714 2016-04-18                  2                 412   
121  4319703577 2016-04-19                  2                 461   
29   1844505072 2016-04-15                  1                 644   
172  4445114986 2016-04-17                  1                  98   
351  7086361926 2016-04-28                  1                 354   
362  7086361926 2016-05-12                  1                 444   
264  5577150313 2016-04-17                  1                 549   
384  8378563200 2016-04-29                  1                 527   
255  5553957443 2016-05-09                  1                 453   

     TotalTimeInBed  Number of days with qualty sleep  
60              568                               NaN  
87              522                               NaN  
121             498                               NaN  
29              961                               NaN  
172             107                               NaN  
351             367                               NaN  
362             457                               NaN  
264             583                               NaN  
384             562                               NaN  
255             494                               NaN

In [39]:
df_activity_weight.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 8 entries, 1503960366 to 8877689391
Data columns (total 20 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   TotalSteps                8 non-null      float64
 1   TotalDistance             8 non-null      float64
 2   TrackerDistance           8 non-null      float64
 3   LoggedActivitiesDistance  8 non-null      float64
 4   VeryActiveDistance        8 non-null      float64
 5   ModeratelyActiveDistance  8 non-null      float64
 6   LightActiveDistance       8 non-null      float64
 7   SedentaryActiveDistance   8 non-null      float64
 8   VeryActiveMinutes         8 non-null      float64
 9   FairlyActiveMinutes       8 non-null      float64
 10  LightlyActiveMinutes      8 non-null      float64
 11  SedentaryMinutes          8 non-null      float64
 12  Calories                  8 non-null      float64
 13  TotalActivityMins         8 non-null      float64
 

In [40]:
#Relationship between Calories and WeightLost
fig = go.Figure()
fig.add_trace(go.Scatter(x=df_activity_weight['Calories'], y=df_activity_weight['WeightLost'], mode='markers', name='WeightLost'))
fig.update_layout(title='Relationship between Calories and WeightLost', xaxis_title='Calories', yaxis_title='WeightLost')
fig.show()


### Data Integrity

**Reliable: Low** 
- accuracy (battery data and sync data, as mentioned in metadata, are not available for analysis)
- bias (MTurk surveys are prone to selection bias, persons willing to be paid to use the device and have their non-identifiable data collected, thus analysis conclusions are not representative of the U.S population as a whole; however, datatypes are objective and provided an efficient way of collecting such data)

**Original: Low**
- data is third-party information, this could be a pro or con, pro being that the company may have been less likely to manipulate data 

**Comprehensive: Med**
- missing information (datasets provide ample data that addresses how device users utilized their device), human error (not a factor in the sense that data collection depends upon device sync with the app, is a factor in the sense that the individual was responsible for monitoring battery life and syncing)

**Current: Low** 
- data from 2016 on device use most likely accurately models how devices would be used in 2022 

**Cited: Low**
- Furberg, R., Brinton, J., Keating, M., & Ortiz, A. (2016). 
- Crowd-sourced Fitbit datasets 03.12.2016-05.12.2016 [Data set]. 
Zenodo. https://doi.org/10.5281/zenodo.53894, the data is cited, however, investigators did not explicitly address how or if limitations in the data where assessed, it is possible that they were assessed by MTurk or the cited investigators, but the process was not expressly explained

Overall, the dataset is considered bad quality data and it is not recommended to produce business recommendations based on this data.

### Data Limitations

1. There' no gender specified. We are assuming the users to be women
2. There are 33 users in data but it's mentioned that there are only 30 users in the metadata.
3. Out of the 8 users entering their weight, 5 users are manually entering their weights at 11:00 pm, whereas the automatic report is syncing at different times of the day. This doesn't seem right

### Conclusion

1. We neeed to work on promoting users to use Bellabeat products to track their sleep.
2. Sedentary users are the majority logging on average 991 minutes or 20 hours making up 81% of total average minutes.
4. Average total steps per day are 7638 which a little bit less for having health benefits for according to the CDC research.
5. We can also see that most users exercise between 5:00 pm and 7:00 pm, so marketing can be concentrated accordingly.
6. There's not enough data to support that Bellabeat helps users in their weightloss journey.
7. Users prefer to track their activities during weekdays as compared to weekends - perhaps because they spend more time outside on weekdays and stay in on weekends.



### Recommendations

1. Bellabeat can improve their Time or leaf product to make it comfortable for users to track their sleep.
2. Analyze their Spring product demand and decide if i's worth promoting. Currently, there's no data collected on hydration.
3. Notify the users about their activity levels to encourage them to workout.